service/src/main/java/bio/terra/pipelines/stairway/imputation/steps/PrepareImputationInputsStep.java

```diff
@@ -116,7 +122,25 @@ public StepResult doStep(FlightContext flightContext) {
 
   @Override
   public StepResult undoStep(FlightContext flightContext) {
-    // no undo for this step
+    // this is the first step in RunImputationGcpJobFlight.
+    // if undoStep is called it means the flight failed
+    // to be moved to a StairwayHook in https://broadworkbench.atlassian.net/browse/TSPS-181
+
+    // set PipelineRun status to FAILED
+    var inputParameters = flightContext.getInputParameters();
+    FlightUtils.validateRequiredEntries(inputParameters, JobMapKeys.USER_ID.getKeyName());
+    pipelineRunsService.markPipelineRunFailed(
+        UUID.fromString(flightContext.getFlightId()),
+        inputParameters.get(JobMapKeys.USER_ID.getKeyName(), String.class));
+
+    // increment failed runs counter metric
```

Comment:
did the gcp flight not have a step taht incremented the failed counter metric before??

Reply:
nope! TSPS-299